src/_zkapauthorizer/tahoe.py

```diff
@@ -404,7 +442,16 @@ def client(self, basedir: Optional[FilePath] = None) -> ITahoeClient:
         return _MemoryTahoe(self, basedir)
 
     def upload(self, data: bytes) -> CapStr:
-        cap = str(self._counter)
+        def encode(s: str) -> str:
+            return b32encode(s.encode("ascii")).decode("ascii")
+
+        cap = "URI:CHK:{}:{}:{}:{}:{}".format(
+            encode("{:016}".format(self._counter)),
+            encode("{:032}".format(self._counter)),
+            self._counter,
+            self._counter,
+            self._counter,
+        )
```

Comment:
I had to add properly-formatted URIs here because of the way the test works for "try to delete a thing from not-a-directory" .. could change the test, I guess, but proper immutable URIs seems like a win.